internal/pkg/diagnostics/diagnostics.go

```diff
@@ -327,12 +327,13 @@ func writeRedacted(errOut, resultWriter io.Writer, fullFilePath string, fileResu
 
 	// Should we support json too?
 	if fileResult.ContentType == "application/yaml" {
-		unmarshalled := map[interface{}]interface{}{}
```

Comment:
I think unmarshaling to `map[interface{}]interface{}` was a restriction with yaml.v2 which is not a part of yaml.v3, see https://github.com/go-yaml/yaml/issues/139#issuecomment-2247444123